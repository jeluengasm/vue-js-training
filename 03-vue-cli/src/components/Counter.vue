<template>
    <h2> Counter </h2>
    <p>{{ counter }}</p>
    <div>
        <span>
            <button @click="increaseButton">+ 1</button> <!-- Same that v-on:click -->
        </span>
        <span>
            <button @click="decreaseButton">- 1</button>
        </span>
    </div>
</template>

<script>
export default {
    name: 'Counter',
    data() {
        return {
        counter: 0
        }
    },
    methods: {
        increaseButton() {
            this.counter++;
        },
        decreaseButton() {
            this.counter--;
        }
    }
}
</script>

<style>
    button {
        background-color: #4CAF50;
        border-radius: 5px;
        border: 1px solid white;
        color: white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover {
        background-color: #3e8e41;
        transition: 0.3s ease-in-out;
    }
</style>