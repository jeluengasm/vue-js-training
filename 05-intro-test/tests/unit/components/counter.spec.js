import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

describe('Counter component', () => {
    test('Must be match with snapshot', () => {
        const wrapper = shallowMount(Counter)

        expect(wrapper.html()).toMatchSnapshot()
    })
    test('h2 must has the default value "Counter"', () => {
        const wrapper = shallowMount(Counter)
        const h2 = wrapper.find('h2')
        expect(h2.exists()).toBeTruthy()
        expect(h2.text()).toBe('Counter')
    })
})