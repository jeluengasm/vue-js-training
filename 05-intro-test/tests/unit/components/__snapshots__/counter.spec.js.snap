// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Must be match with snapshot 1`] = `
<h2>Counter</h2>
<p>100</p>
<div><span><button>+ 1</button><!-- Same as v-on:click --></span><span><button>- 1</button></span></div>
`;
